warp10/src/main/java/io/warp10/script/filler/FillerInterpolate.java

```diff
@@ -109,4 +115,111 @@ public int getPreWindow() {
   public int getPostWindow() {
     return 1;
   }
+  
+  public WarpScriptSingleValueFillerFunction compute(GeoTimeSerie gts) throws WarpScriptException {
+    
+    final PolynomialSplineFunction valFunction;
+    final PolynomialSplineFunction latFunction;
+    final PolynomialSplineFunction lonFunction;
+    final PolynomialSplineFunction elevationFunction;
+
+    if (gts.size() == 0) {
+      valFunction = null;
+      latFunction = null;
+      lonFunction = null;
+      elevationFunction = null;
+    } else {
+      if (GeoTimeSerie.TYPE.DOUBLE != gts.getType() && GeoTimeSerie.TYPE.LONG != gts.getType()) {
+        throw new WarpScriptException(getName() + " expects a GTS of type DOUBLE or LONG, but instead got a GTS of type " + gts.getType().name());
+      }
+
+      // values
+      double[] xval = GTSHelper.getTicksAsDouble(gts);
+      double[] fval = GTSHelper.getValuesAsDouble(gts);
+
+      int size = gts.size();
+      if (size > 1) {
+        valFunction = (new LinearInterpolator()).interpolate(xval, fval);
+      } else {
+        valFunction = null;
+      }
+
+      // positions
+      // how many ticks do have a location ?
+      int nLoc = 0;
+      for (int i = 0; i < gts.size(); i++) {
+        if (GeoTimeSerie.NO_LOCATION != GTSHelper.locationAtIndex(gts, i)) {
+          nLoc++;
+        }
+      }
+      if (nLoc > 0) {
+        double[] xLocTicks = new double[nLoc];
+        double[] fLatVal = new double[nLoc];
+        double[] fLonVal = new double[nLoc];
+        int idx = 0;
+        for (int i = 0; i < gts.size(); i++) {
+          long l = GTSHelper.locationAtIndex(gts, i);
+          if (l != GeoTimeSerie.NO_LOCATION) {
+            double[] latlon = GeoXPLib.fromGeoXPPoint(l);
+            xLocTicks[idx] = GTSHelper.tickAtIndex(gts, i);
+            fLatVal[idx] = latlon[0];
+            fLonVal[idx] = latlon[1];
+            idx++;
+          }
+        }
+        latFunction = (new LinearInterpolator()).interpolate(xLocTicks, fLatVal);
```

Comment:
This is a breaking change. Old filler.interpolate does not handle geo filling based on the previous and next existing geo coordinate.
See the apply method of the same class used in cross FILL.
If the previous data point does not have a geo coordinate, then the actual data point would not get a geo coordinate. In your implementation, it can if the datapoint before the previous has geo coordinates.

Reply:
Indeed. You will have more datapoints filled with geo information than before with this new implementation.

Reply:
As discussed IRL, if a customer find this edge case with filler.interpolate, it can switch to the map signature, and add `'force.old.filler' true`  flag. Execution time is then multiplied by 10.